sklearn/model_selection/_split.py

```diff
@@ -913,6 +915,243 @@ def get_n_splits(self, X, y, groups):
         return int(comb(len(np.unique(groups)), self.n_groups, exact=True))
 
 
+class _RepeatedSplits(with_metaclass(ABCMeta)):
+    """Repeated splits for K-Fold and Stratified K-Fold
+
+    Repeats splits for cross-validators n times.
+
+    Parameters
+    ----------
+    cv : object, KFold or StratifiedKFold
+        Instance of cross-validator.
+
+    n_repeats : int, default=5
+        Number of times cross-validator needs to be repeated.
+
+    random_state : None, int or RandomState, default=None
+        Random state to be used to generate random state for each
+        repetition.
+    """
+    def __init__(self, cv, n_repeats=5, random_state=None):
+        if not isinstance(cv, (KFold, StratifiedKFold)):
+            raise ValueError(
+                "cv must be an instance of KFold or StratifiedKFold.")
+
+        if not isinstance(n_repeats, (np.integer, numbers.Integral)):
+            raise ValueError("Number of repetitions must be of Integral type.")
+
+        if n_repeats <= 1:
+            raise ValueError("Number of repetitions must be greater than 1.")
+
+        rng = check_random_state(random_state)
```

Comment:
we are not certain it's the best design, but currently all the splitters do this in `split`, not in `__init__`